src/lib.rs

```diff
@@ -604,7 +604,12 @@ impl DTVCCPacket {
                 service.len() - self.free_space(),
             ));
         }
-        self.services.push(service);
+
+        // Services allowed are either the padding 0 service or non-empty
+        // services
+        if service.number == 0 || !service.codes().is_empty() {
+            self.services.push(service);
```

Comment:
Should an error be returend when the provided service is not used?

Reply:
Good point. It maybe should to inform the caller that it's doing something wrong. Is there an appropriate error type to use ?

Reply:
The error type is already a `WriterError`.  You may need to add another value to that enum for this.